### PR TITLE
test new split punc pdfplumber parser

### PR DESCRIPTION
**Pull request by kyleclo**

### CODE REVIEW

File: tests/test_parsers/test_pdf_plumber_parser.py [hunk 1]
```diff
@@ -24,4 +25,27 @@ def test_parse(self):
...
+        assert '[1-5]' in no_split_tokens_with_numbers
+        assert 'GPT-3[10]' in no_split_tokens_with_numbers
+
+        split_parser = PDFPlumberParser(split_at_punctuation=True)
+        split_doc = split_parser.parse(input_pdf_path="../fixtures/2107.07170.pdf")
+        split_tokens_with_numbers = [token.text for token in split_doc.tokens if re.search(r'[0-9]', token.text)]
+        assert '[1-5]' not in split_tokens_with_numbers
+        assert '1-5' in split_tokens_with_numbers
+        assert 'GPT-3[10]' not in split_tokens_with_numbers
+        assert 'GPT-3' in split_tokens_with_numbers
```
Comment (geli-gel) [inline tests/test_parsers/test_pdf_plumber_parser.py:42]: so, does it split like: "GPT-3", "[", "10", "]" ?

File: tests/test_parsers/test_pdf_plumber_parser.py [hunk 2]
```diff
@@ -24,4 +25,27 @@ def test_parse(self):
...
+
+        custom_split_parser = PDFPlumberParser(split_at_punctuation='!"#$%&\'()*+,-./:;<=>?@[\\]^_`{|}~–§')
+        custom_split_doc = custom_split_parser.parse(input_pdf_path="../fixtures/2107.07170.pdf")
+        custom_split_tokens_with_numbers = [token.text for token in custom_split_doc.tokens if re.search(r'[0-9]', token.text)]
+        assert '1-5' not in custom_split_tokens_with_numbers
+        assert 'GPT-3' not in custom_split_tokens_with_numbers
```
Comment (geli-gel) [inline tests/test_parsers/test_pdf_plumber_parser.py:48]: and this one splits like "GPT", "-", "3", "[", "10", "]" ?

Comment (kyleclo) [reply tests/test_parsers/test_pdf_plumber_parser.py:48]: yes that's right

File: tests/test_parsers/test_pdf_plumber_parser.py (hunk 1 above)
Comment (kyleclo) [reply tests/test_parsers/test_pdf_plumber_parser.py:42]: yep!

**Review by geli-gel — APPROVED** [review]

**Pull request merged**